modules/nix/default.nix

```diff
@@ -761,6 +763,13 @@ in
         { assertion = elem "nixbld" config.users.knownGroups -> elem "nixbld" createdGroups; message = "refusing to delete group nixbld in users.knownGroups, this would break nix"; }
         { assertion = elem "_nixbld1" config.users.knownGroups -> elem "_nixbld1" createdUsers; message = "refusing to delete user _nixbld1 in users.knownUsers, this would break nix"; }
         { assertion = config.users.groups ? "nixbld" -> config.users.groups.nixbld.members != []; message = "refusing to remove all members from nixbld group, this would break nix"; }
+
+        {
+          # Should be fixed in Lix by https://gerrit.lix.systems/c/lix/+/2100
+          # As `isNixAtLeast "2.92.0" "2.92.0-devpre20241107" == false`, we need to explicitly check if the user is running Lix 2.92.0
+          assertion = cfg.settings.auto-optimise-store -> (cfg.package.pname == "lix" && (isNixAtLeast "2.92.0-devpre20241107" || cfg.package.version == "2.92.0"));
+          message = "`auto-optimise-store` is known to corrupt the Nix Store, please use `nix.optimise.automatic` instead.";
```

Comment:
I haven't worked in this repo really, but I'm curious how these assertions are scoped in the output. Will it be obvious where the `auto-optimise-store` is? Or would it be beneficial to make it more explicit with `nix.settings.auto-optimise-store` or (nix.settings): `auto-optimise-store`? Just some thoughts.

Reply:
It shows up as:

```
       error:
       Failed assertions:
       - `auto-optimise-store` is known to corrupt the Nix Store, please use `nix.optimise.automatic` instead.
```

Changed it to be more explicit 👍

Reply:
I doubt it would be a problem in practice as users would be able to search their configs for `auto-optimise-store` but making it more explicit will probably be for the best